framework/src/ics/InitialConditionBase.C

```diff
@@ -28,9 +30,13 @@ InitialConditionBase::validParams()
                         "When set to true, a UserObject retrieved "
                         "by this IC will not be executed before the "
                         "this IC");
-
   params.addParamNamesToGroup("ignore_uo_dependency", "Advanced");
 
+  MooseEnum stateEnum("CURRENT=0 OLD=1 OLDER=2", "CURRENT");
+  params.addParam<MooseEnum>("state", stateEnum, "Specify state that is being set.");
+
+  params.addParamNamesToGroup("state", "Advanced");
```

Comment:
I feel like this doesn't have to go into the advanced group, but this is apparently debatable.